python/cuml/dask/linear_model/linear_regression.py

```diff
@@ -311,53 +192,36 @@ def predict(self, X):
         -------
         y : dask cuDF (n_rows, 1)
         """
-        gpu_futures = self.client.sync(extract_ddf_partitions, X)
-
-        worker_to_parts = OrderedDict()
-        for w, p in gpu_futures:
-            if w not in worker_to_parts:
-                worker_to_parts[w] = []
-            worker_to_parts[w].append(p)
+        X = X.persist()
 
-        key = uuid1()
-        partsToSizes = [(self.rnks[wf[0]], self.client.submit(
-            LinearRegression._func_get_size,
-            wf[1],
-            workers=[wf[0]],
-            key="%s-%s" % (key, idx)).result())
-            for idx, wf in enumerate(gpu_futures)]
-
-        n_cols = X.shape[1]
-        n_rows = reduce(lambda a, b: a+b, map(lambda x: x[1], partsToSizes))
+        data = MGData.single(X, client=self.client)
 
         key = uuid1()
-        linear_pred = dict([(self.rnks[wf[0]], self.client.submit(
+        linear_pred = dict([(wf[0], self.client.submit(
```

Comment:
API is definitely cleaner with this. Per other discussion, I'm definitely interested in seeing if we could do this with map_partitions somehow instead.

Reply:
Dante & I have been discussion having a `lazy` option for all the model classes `model.predict(X, lazy=True)` for now. Of course it would be well-documented as to the fact that it doesn't currently share any state so the amount of storage needed for the models is going to be equal to the number of partitions * size of model params. For now, this will enable users a path to truly lazy operation, at least until we can find a good way to share the model state and possibly even limit the number of concurrent predictions on each worker, in a truly lazy way. What do you think of this solution, at least for the short term?

Reply:
cc @Salonijain27 who is also looking into this